backend/article/tasks.py

```diff
@@ -0,0 +1,143 @@
+"""
+Asynchronous tasks for fetching articles using celery
+"""
+
+# from celery import task
+from celery import shared_task
+from celery.utils.log import get_task_logger
+from article.models import Article
+from terminator import sensitive
+import requests
+from bs4 import BeautifulSoup
+import logging
+from gensim.summarization import keywords
+from nltk.stem import WordNetLemmatizer
+from wordlist.models import Phrase, Word
+import nltk
+
+logger = logging.getLogger(__name__)
+
+@shared_task
+def fetch_article_nytimes():
+  """
+    Fetch today's most popular news and
+    Save it as **Article** model.
+    This task can be scheduled via either 
+    Django admin console or settings.py
+  """
+  try:
+    url = r"https://api.nytimes.com/svc/mostpopular/v2/viewed/1.json?api-key=%s" % sensitive.NYTIMES_API_KEY
+    response = requests.get(url)
+    data = response.json()
+
+    num = data['num_results']
+    results = data['results']
+
+    logger.debug('Fetching articles from NYTimes...')
+    for result in results:
+      response = requests.get(result['url'])
+      original_url = result['url']
+      title = result['title']
+      author = result['byline']
+      if author.startswith('By '):
+        author = author[3:]
+      soup = BeautifulSoup(response.text, 'html.parser')
+      category = result['section']
+      subcategory = result['subsection']
+
+      logger.debug('Fetching from: %s' % original_url)
+      logger.debug('Title: %s' % title)
+      logger.debug('Author: %s' % author)
+
+      article = soup.select('article')[0]
+      section = article.select('section')[-1]
+      text = section.text
+      logger.debug('Content: %s' % text)
+
+      phrases, keywords = extract_phrases_and_keywords(text, 10)
+
+      article_model = Article(title=title, author=author, content=text)
+      article_model.save()
+
+      for (phrase, keyword) in zip(phrases, keywords):
+        # TODO: Change korean meaning to real korean meaning
+        word_model = Word(content=keyword, korean_meaning="가짜뜻", difficulty=5)
+        word_model.save()
+        phrase_model = Phrase(content=phrase, word=word_model)
+        phrase_model.save()
+        article_model.phrases.add(phrase_model.id)
+
+    logger.debug('Fetching Completed!')
+
+  except Exception as e:
+    print("Something went wrong: ", repr(e))
+
+
+def extract_phrases_and_keywords(content, number):
+  """
+    Return list of extracted phrases and keywords from content
+    :param content: content from which phrases and keywords are extracted
+    :type content: str
+    :param number: number of phrases and keywords returned
+    :type number: int
+    :returns: (a list of phrases, a list of keywords)
+    :rtype: tuple
+  """
+  keywords, l_keywords = extract_keywords(content, number, get_both=True)
```

Comment:
l_keyword is a little bit confusing, if not used through various files, how about making variable names specific? (even if they get long)

Reply:
Resolved Now, l_keyword -> keyword, keyword -> raw_keyword.